src/modals/new-identity-flow/NewIdentityFlow.js

```diff
@@ -86,54 +94,114 @@ class NewIdentityFlow extends Component {
 
-        this.setState({ currentFlow: flow, currentStepId: createIdentityFirstStepId });
-    };
+        const identityInfo = data['create-identity-info'];
+        const deviceInfo = data['create-identity-device'];
 
-    handleChooseBackupFlow = () => {
-        console.log('Go to Backup Identity Flow');
-    };
+        const profileDetails = {
+            '@context': 'https://schema.org',
+            '@type': identityInfo.type,
+            name: identityInfo.name,
+        };
+
+        // Read image from File object to an ArrayBuffer
+        if (identityInfo.image) {
+            profileDetails.image = {
+                type: identityInfo.image.type,
+                data: await readAsArrayBuffer(identityInfo.image),
+            };
+        }
+
+        return createIdentity({
+            profileDetails,
+            deviceInfo,
+        });
+    }
+
+    importIdentity(data) {
+        console.log('data', data);
+
+        return Promise.reject(new Error('Not implemented'));
+    }
+
+    createFlowPromise = (currentStepId, data) => {
+        const { currentFlow } = this.state;
+        const isLastCreateStep = currentFlow === 'create' && currentStepId === 'create-identity-feedback';
 
-    handleRetrySubmit = () => {
-        console.log('Retry form submission');
+        if (isLastCreateStep) {
+            return this.createIdentity(data);
+        }
+
+        const isLastImportStep = currentFlow === 'import' && currentStepId === 'import-identity-feedback';
+
+        if (isLastImportStep) {
+            return this.importIdentity(data);
+        }
     };
 
-    handleNextStep = (nextStepId, data) => {
-        this.setState((prevState) => ({
-            currentStepId: nextStepId,
-            data: { ...prevState.data, [prevState.currentStepId]: data },
-        }));
+    handleExited = () => {
+        this.setState(initialState);
+        this.props.onExited && this.props.onExited();
     };
 
-    handleCreateSubmitForm = (nextStepId, data) => {
-        // Skip if there's a ongoing promise
-        if (getPromiseState(this.state.promise).status !== 'none') {
-            return;
-        }
+    handleChooseFlow = (flow) => {
+        const createIdentityFirstStepId = flow === 'create' ? 'create-identity-info' : 'import-identity-mnemonic';
 
-        const promise = new Promise((resolve) => setTimeout(resolve, 5000));
-        const finalData = { ...this.state.data, [this.state.currentStepId]: data };
+        this.setState({ currentFlow: flow, currentStepId: createIdentityFirstStepId });
+    };
 
-        this.setState({
-            currentStepId: nextStepId,
-            data: finalData,
-            promise,
+    handleNextStep = (nextStepId, currentStepData) => {
+        this.setState((prevState) => {
+            const data = { ...prevState.data, [prevState.currentStepId]: currentStepData };
+            const currentStepId = nextStepId;
+            const promise = this.createFlowPromise(currentStepId, data);
```

Comment:
This change was because of two reasons:
- Avoid a unnecessary rerender 
- To circumvent a big of the modal that was not rendering the last step (the bug is still there, but happens rarely with this)

//cc @PedroMiguelSS

Reply:
We will merge this as is and wait for @PedroMiguelSS to fix the bug.